src/main/java/de/tum/in/www1/artemis/web/rest/UserJWTController.java

```diff
@@ -102,43 +103,29 @@ public ResponseEntity<JWTToken> authorizeSAML2(@RequestBody final String body) {
 
         try {
             authentication = saml2Service.get().handleAuthentication(principal);
+            SecurityContextHolder.getContext().setAuthentication(authentication);
         }
         catch (UserNotActivatedException e) {
             // If the exception is not caught a 401 is returned.
             // That does not match the actual reason and would trigger authentication in the client
             return ResponseEntity.status(HttpStatus.FORBIDDEN).header("X-artemisApp-error", e.getMessage()).build();
         }
 
-        final String jwt = tokenProvider.createToken(authentication, rememberMe);
-        final HttpHeaders httpHeaders = new HttpHeaders();
-        httpHeaders.add(JWTFilter.AUTHORIZATION_HEADER, "Bearer " + jwt);
-        return new ResponseEntity<>(new JWTToken(jwt), httpHeaders, HttpStatus.OK);
+        final boolean rememberMe = Boolean.parseBoolean(body);
+        ResponseCookie responseCookie = jwtCookieService.buildLoginCookie(rememberMe);
+        response.addHeader(HttpHeaders.SET_COOKIE, responseCookie.toString());
+
+        return ResponseEntity.ok().build();
     }
 
     /**
-     * Object to return as body in JWT Authentication.
+     * Removes the cookie containing the jwt
+     * @param response  HTTP response
      */
-    public static class JWTToken {
-
-        private String idToken;
-
-        /**
-         * Make Jackson happy
-         */
-        JWTToken() {
-        }
-
-        JWTToken(String idToken) {
-            this.setIdToken(idToken);
-        }
-
-        @JsonProperty("id_token")
-        public String getIdToken() {
-            return idToken;
-        }
-
-        void setIdToken(String idToken) {
-            this.idToken = idToken;
-        }
+    @PostMapping("/logout")
+    public void logout(HttpServletResponse response) {
+        // Logout needs to build the same cookie (secure, httpOnly and sameSite='Lax') or browsers will ignore the header and not unset the cookie
+        ResponseCookie responseCookie = jwtCookieService.buildLogoutCookie();
+        response.addHeader(HttpHeaders.SET_COOKIE, responseCookie.toString());
```

Comment:
I've encountered Issues with the SAML2 workflow. After logout it was not possible to login again.

This might work:

```java
        // Logout needs to build the same cookie (secure, httpOnly and sameSite='Lax') or browsers will ignore the header and not unset the cookie
        ResponseCookie jwtResetCookie = jwtCookieService.buildLogoutCookie();
        response.addHeader(HttpHeaders.SET_COOKIE, jwtResetCookie.toString());
        // Session has to be destroyed for SAML2 Workflow to work
        ResponseCookie jsessionResetCookie = ResponseCookie.from("JSESSIONID", "").path("/").httpOnly(true).secure(true).maxAge(0).build();
        response.addHeader(HttpHeaders.SET_COOKIE, jsessionResetCookie.toString());

```